src/SessionSetup/OrbitServiceInstance.cpp

```diff
@@ -0,0 +1,138 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "SessionSetup/OrbitServiceInstance.h"
+
+#include <absl/strings/str_format.h>
+
+#include <QCoreApplication>
+#include <QMetaEnum>
+#include <QProcess>
+#include <filesystem>
+#include <memory>
+#include <string>
+
+#include "OrbitBase/Logging.h"
+#include "OrbitBase/Result.h"
+
+namespace orbit_session_setup {
+
+constexpr int kWaitTimeInMilliSeconds = 2000;
+
+class OrbitServiceInstanceImpl : public OrbitServiceInstance {
+ public:
+  explicit OrbitServiceInstanceImpl(const QString& program, const QStringList& arguments);
+  ~OrbitServiceInstanceImpl() override;
+
+  [[nodiscard]] bool IsRunning() const override;
+  [[nodiscard]] ErrorMessageOr<void> Shutdown() override;
+
+  [[nodiscard]] ErrorMessageOr<void> Start();
+
+ private:
+  [[nodiscard]] std::string GetStateErrorAndOutput();
```

Comment:
We usually use `Get` for non-mutating operations. I would prefer something that emphasizes the mutating aspect, maybe `ReadStateErrorAndOutput`. (But this is not a strong opinion)

Reply:
I like it